EIPS/eip-3074.md

```diff
@@ -0,0 +1,159 @@
+ - `to` - the address of the callee (not `CALL_PRECOMPILE`);
+ - `gaslimit` - the minimum gas limit which must be provided with the call into `CALL_PRECOMPILE`;
+ - `value` - the exact amount of Ether in wei to transfer from the relayer to the callee;
+ - `data` - the calldata for the call into `to`; and
+ - `v`, `r`, `s` - signature for the package, including chain id as specified in [EIP-115](./eip-115.md).
+
+The arguments to `CALL_PRECOMPILE` are encoded as `rlp([nonce, gaslimit, to, value, data, v, r, s])`.
+
+The signature (`v`, `r`, `s`) arguments are computed from `secp256k1(keccak256(rlp([nonce, gaslimit, to, value, data, relayer])))`.
+
+`CALL_PRECOMPILE` returns a failure without changing the nonce in the following situations:
+ - Invalid signature
+ - Future or past nonce
+ - Gas limit supplied with the call into `CALL_PRECOMPILE` is less than the signed `gaslimit`
+ - The relayer's balance is insufficient to pay for the supplied gas plus `value`
+
+`CALL_PRECOMPILE` returns a success in all other cases.
+
+The return data of `CALL_PRECOMPILE` will be a single byte to indicate the status of the call into callee followed immediately by the return data from that call.
+
+#### `NONCE_PRECOMPILE`
+
+`NONCE_PRECOMPILE` requires the following two arguments:
+
+ - `relayer` - the relayer address; and
+ - `sponsee` - the sponsee address.
+
+Assuming the calldata is the correct length, `NONCE_PRECOMPILE` will return a success, and place the nonce associated with the address pair in the return data.
+
+### Nonces
+
+The two precompiles will maintain a nonce for each pair of relayer address and sponsee address, in essence:
+
+```
+{
+    (0x1234...5678, 0xEEEE...EEEE) => 55,
+    (0x1122...3344, 0xBBBB...BBBB) => 89,
+}
+```
+
+Where:
+ * `0x1234...5678` and `0x1122...3344` are the relayer addresses;
+ * `0xEEEE...EEEE` and `0xBBBB...BBBB` are the sponsee addresses; and
+ * `55` and `89` are the current nonce values for their respective pairs.
+
+The nonce shall be incremented whenever a correctly signed transaction-like package containing the next nonce is submitted to `CALL_PRECOMPILE` with a sufficient gas limit.
+
+### Gas Fees
+
+#### `CALL_PRECOMPILE`
+
+TODO: Probably something like the sum of:
+ * The cost of a normal call, including calldata and signature size, etc.
+ * An `SLOAD` to read the current nonce
+ * An `SSTORE` to write the updated nonce
+ * Cost of an ecrecover
+
+#### `NONCE_PRECOMPILE`
+
+TODO: Probably something like the sum of:
+ * An `SLOAD` to read the current nonce
+
+## Rationale
+
+### Another Sponsored Transaction EIP
+
+Other approaches to sponsored transactions, like [EIP-2711](./eip-2711.md) and [EIP-2733](./eip-2733.md), are incompatible with [EIP-2938](./eip-2938.md) flavored account abstraction (AA). These proposals require a _signed_ transaction from the sponsor's account, which is not possible from an AA contract, because it has no private key to sign with.
```

Comment:
You may want to consider not including references to other competing draft EIPs in this EIP as it creates a dependency that you probably don't want.  It would be unfortunate if this EIP wasn't able to go to final because it points at draft EIPs internally!  Instead, consider just *briefly* describing the alternatives (like a sentence or two) inline and then describing why this alternative is better.  For example:
> We could have done XXX but that would not work with any future account abstraction proposals.